DECLARING_DEPENDENCIES.md

```diff
@@ -0,0 +1,210 @@
+# Declaring Dependencies on Google Java Libraries
+
+Google maintains a number of open source Java libraries that make it
+easier to use services in Google Cloud Platform (GCP). Additionally Google
+maintains several foundational libraries that can be used for very
+general purposes, which the GCP libraries also depend on. A user of
+these libraries needs to use compatible versions of them in order to
```

Comment:
"A user of
these libraries needs to use compatible versions of them in order to
avoid dependency conflicts. This document explains how to use BOMs
(bill of materials) to accomplish that and explains additional
practices that make dependency management go more smoothly" 

could be cut to just

This document explains how to use bills of materials (BOMs) to avoid dependency conflicts.

Reply:
But this document isn't *just* about using BOMs. It already has a recommendation for `requireUpperBoundDeps`, and later I'd also like to add a recommendation for using beta checkers. I will make this more concise, though.